### PR TITLE
Fix error on state file FAQ page by adding filing years to NJ

### PR DESCRIPTION
**Pull request by jenny-heath**
## Link to Sentry issue
- https://codeforamerica.sentry.io/issues/5827535300/?alert_rule_id=1046555&alert_type=issue&notification_uuid=a00e7a97-90ae-469c-9b42-8a110921010c&project=1880327&referrer=slack
## Is PM acceptance required? (delete one)
- Yes - don't merge until JIRA issue is accepted!
## What was done?
- like the title says
- also:
  - refactored state info service and specs so that they test all the states
  - refactored the schema file loader because it doesn't seem like we need an attribute on the info service just for that but maybe i'm missing something?
## How to test?
- one reservation about the info service tests: it doesn't feel like we *need* to require all the methods be defined but it also felt like overkill to separate them into required and not required. open to suggestions!
## Screenshots (for visual changes)
- Before
image

- After
image

### CODE REVIEW

**Review by mrotondo — APPROVED** [review]

File: app/services/state_file/state_information_service.rb [hunk 1]
```diff
@@ -101,23 +103,24 @@ def state_code_to_name_map
...
+                              "Processing Center<br/>" \
+                              "Trenton, NJ".html_safe,
+        pay_taxes_link: "https://www.nj.gov/treasury/taxation/payments-notices.shtml",
+        survey_link: "",
+        tax_payment_info_url: "https://www.nj.gov/treasury/taxation/payments-notices.shtml",
+        tax_payment_url: "https://www.nj.gov/treasury/taxation/payments-notices.shtml",
+        tax_refund_url: "https://www.tax.ny.gov/pit/file/refund.htm",
+        vita_link: "",
+        voucher_form_name: "NJ Voucher Form",
+        voucher_path: "/pdfs/it201v_1223.pdf", # TODO
```
Comment (jachan) [inline app/services/state_file/state_information_service.rb:116]: [DUST] Is it possible to leave voucher_path commented? Or at least blank if not? Concerned about pointing to a nonexistent path.

**Review by jachan — COMMENTED** [review]
Had a question about the NJ Voucher Path - otherwise looks good! Thanks for taking the initiative to help fix the FAQ page!

File: app/services/state_file/state_information_service.rb (hunk 1 above)
Comment (aloverso) [reply app/services/state_file/state_information_service.rb:116]: Or we could set it to an empty string?

**Review by aloverso — APPROVED** [review]

File: app/services/state_file/state_information_service.rb (hunk 1 above)
Comment (jenny-heath) [reply app/services/state_file/state_information_service.rb]: done!

**Pull request merged**